documentcloud/addons/models.py

```diff
@@ -577,6 +582,44 @@ def dispatch(self, document_pk=None):
             )
 
 
+class AddOnDisableLog(models.Model):
+    """Log usage of Add-On Disables"""
+
+    # Reference to the AddOnEvent
+    addon_event = models.ForeignKey(
+        verbose_name=_("event reference"),
+        to="addons.AddOnEvent",
+        on_delete=models.PROTECT,
+        related_name="disable_logs",
+        help_text=_("Reference to the Add-On Event"),
+    )
+
+    event = models.IntegerField(
+        _("event"),
+        choices=Event.choices,
+        help_text=_("The current event state of the Add-On"),
+    )
+    created_at = AutoCreatedField(
+        _("created at"),
+        help_text=_("Timestamp of when the add-on disable log was created"),
+        db_index=True,
+    )
+    previous_event_state = models.IntegerField(
+        _("previous event state"),
+        choices=Event.choices,
+        help_text=_("The event state of the Add-On before disable"),
+    )
+
+    def revert_event(self):
+        # Logic to revert the event
+        addon_event = self.addon_event
+        addon_event.event = self.previous_event_state
+        addon_event.save()
```

Comment:
I am thinking if we might want to add a flag to track if the log has been reverted or not, just for record keeping